subsys/bluetooth/audio/shell/bap_broadcast_assistant.c

```diff
@@ -493,13 +499,36 @@ static int cmd_bap_broadcast_assistant_add_src(const struct shell *sh,
 
 static bool broadcast_source_found(struct bt_data *data, void *user_data)
 {
+	static char identified_name_addr_str[BT_ADDR_LE_STR_LEN];
 	struct bt_bap_broadcast_assistant_add_src_param param = { 0 };
 	const struct bt_le_scan_recv_info *info = user_data;
 	char addr_str[BT_ADDR_LE_STR_LEN];
 	struct bt_uuid_16 adv_uuid;
 	uint32_t broadcast_id;
+	bool identified_name_matched_addr = false;
 	int err;
 
+	/* Check for matched broadcast name (in case of add_broadcast_name) */
+	if (data->type == BT_DATA_BROADCAST_NAME) {
+		if (strlen(auto_scan.broadcast_name) != 0) {
+			char broadcast_name[BT_AUDIO_BROADCAST_NAME_LEN_MAX + 1] = { 0 };
+			size_t data_length = MIN(data->data_len, sizeof(broadcast_name));
```

Comment:
If `data->data_len` > BT_AUDIO_BROADCAST_NAME_LEN_MAX` or `data->data_len < BT_AUDIO_BROADCAST_NAME_LEN_MIN` it's invalid.

We never need to truncate it

Reply:
Ok, I'll correct it.